### PR TITLE
Release 0.2.1 doc changes

### PR DESCRIPTION
**Pull request by awholegunch**
Documentation changes for the 0.2.1 release.
Main change: release notes
Minor changes: 
- Fixed links, added an "all" target to the build.
- Added automatic versioning in the "Getting Started" guide, so that the code samples and links match the build version.
- Modified the FAQ, to improve formatting and content.
- Added the new "ClickStream" app as a reference in the examples section.

Staged at: http://stg-web101.sw.joyent.continuuity.net/tigon/0.2.1-r0.2.1_docs/en/release-notes.html

### CODE REVIEW

**Comment by achalpandey** [comment]
LGTM :+1:

**Comment by gokulavasan** [comment]
@awholegunch Can we change ClickStream to ClickStreamFlow. I think that is the only change required?

File: tigon-docs/developer-guide/source/examples/index.rst [hunk 1]
```diff
@@ -49,10 +49,14 @@ Reference applications built with Tigon are included in the
 - `AdNetworkFlow: <https://github.com/caskdata/tigon-apps/tree/develop/AdNetworkFlow>`__
   Demonstrates using Tigon to write a realtime bidding (RTB) advertisement framework.
 
+- `ClickStream: <https://github.com/caskdata/tigon-apps/tree/develop/ClickStream>`__
```
Comment (gokulavasan) [inline tigon-docs/developer-guide/source/examples/index.rst]: ClickStreamFlow

**Comment by gokulavasan** [comment]
LGTM :+1:

**Pull request merged**